http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -974,4 +910,82 @@ object LeonardoService {
     }
+
+  private[service] def augmentCreateRuntimeRequest(serviceAccountInfo: ServiceAccountInfo,
+                                                   googleProject: GoogleProject,
+                                                   clusterName: RuntimeName,
+                                                   userEmail: WorkbenchEmail,
+                                                   request: CreateRuntimeRequest,
+                                                   clusterImages: Set[RuntimeImage]): CreateRuntimeRequest = {
+    val userJupyterExt = request.jupyterExtensionUri match {
+      case Some(ext) => Map("notebookExtension" -> ext.toUri)
+      case None      => Map.empty[String, String]
+    }
+
+    // add the userJupyterExt to the nbExtensions
+    val updatedUserJupyterExtensionConfig = request.userJupyterExtensionConfig match {
+      case Some(config) => config.copy(nbExtensions = config.nbExtensions ++ userJupyterExt)
+      case None         => UserJupyterExtensionConfig(userJupyterExt, Map.empty, Map.empty, Map.empty)
+    }
+
+    // transform Some(empty, empty, empty, empty) to None
+    val updatedRequest = request.copy(
+      userJupyterExtensionConfig =
+        if (updatedUserJupyterExtensionConfig.asLabels.isEmpty)
+          None
+        else
+          Some(updatedUserJupyterExtensionConfig)
+    )
+
+    addClusterLabels(serviceAccountInfo, googleProject, clusterName, userEmail, updatedRequest, clusterImages)
+  }
+
+  private[service] def calculateAutopauseThreshold(autopause: Option[Boolean],
+                                                   autopauseThreshold: Option[Int],
+                                                   autoFreezeConfig: AutoFreezeConfig): Int =
+    autopause match {
+      case None =>
+        autoFreezeConfig.autoFreezeAfter.toMinutes.toInt
+      case Some(false) =>
+        autoPauseOffValue
+      case _ =>
+        if (autopauseThreshold.isEmpty) autoFreezeConfig.autoFreezeAfter.toMinutes.toInt
+        else Math.max(autoPauseOffValue, autopauseThreshold.get)
+    }
+
+  private[service] def addClusterLabels(serviceAccountInfo: ServiceAccountInfo,
+                                        googleProject: GoogleProject,
+                                        clusterName: RuntimeName,
+                                        creator: WorkbenchEmail,
+                                        request: CreateRuntimeRequest,
+                                        clusterImages: Set[RuntimeImage]): CreateRuntimeRequest = {
+    // create a LabelMap of default labels
+    val defaultLabels = DefaultLabels(
+      clusterName,
+      googleProject,
+      creator,
+      serviceAccountInfo.clusterServiceAccount,
+      serviceAccountInfo.notebookServiceAccount,
+      request.jupyterUserScriptUri,
+      request.jupyterStartUserScriptUri,
+      clusterImages.map(_.imageType).filterNot(_ == Welder).headOption
+    )
+
+    val defaultLabelsMap = defaultLabels.asJson.asObject
```

Comment:
I see you added a `DefaultLabels.toMap` -- use that instead instead of json encoding to build a map?

Reply:
ah, this was addressed previously...bad merge